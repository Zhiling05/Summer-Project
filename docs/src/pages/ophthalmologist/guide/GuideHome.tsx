const GuideHome = () => {
  return <div>Guide Home Page</div>;
};
export default GuideHome;
