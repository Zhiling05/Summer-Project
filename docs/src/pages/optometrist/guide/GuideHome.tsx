import { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import Header from '../../../components/Header';
import BottomNav from '../../../components/BottomNav';
import '../../../styles/card.css';

export default function GuideHome() {
  const [selectedCard, setSelectedCard] = useState<string | null>(null);
  const navigate = useNavigate();

  const handleCardClick = (cardType: string) => {
    setSelectedCard(cardType);
    console.log(`Selected: ${cardType}`);
    if (cardType === 'gallery') {
      navigate('gallery');
    } else if (cardType === 'tutorial') {
      navigate('tutorial');
    }
  };

  const cards = [
    {
      id: 'gallery',
      title: 'Reference Image Gallery',
      description: 'Visual references and diagnostic imaging examples for assessment',
      icon: (
        <svg 
          width="16" 
          height="16" 
          viewBox="0 0 24 24" 
          fill="none" 
          stroke="white" 
          strokeWidth="2"
          strokeLinecap="round" 
          strokeLinejoin="round"
        >
          <rect x="3" y="3" width="18" height="18" rx="2" ry="2"/>
          <circle cx="8.5" cy="8.5" r="1.5"/>
          <path d="M21 15l-5-5L5 21"/>
        </svg>
      )
    },
    {
      id: 'tutorial',
      title: 'App Tutorial for Optometrists',
      description: 'Learn how to navigate and use all features',
      icon: (
        <svg 
          width="16" 
          height="16" 
          viewBox="0 0 24 24" 
          fill="none" 
          stroke="white" 
          strokeWidth="2"
          strokeLinecap="round" 
          strokeLinejoin="round"
        >
          <path d="M12 6.253v13m0-13C10.832 5.477 9.246 5 7.5 5S4.168 5.477 3 6.253v13C4.168 18.477 5.754 18 7.5 18s3.332.477 4.5 1.253m0-13C13.168 5.477 14.754 5 16.5 5c1.747 0 3.332.477 4.5 1.253v13C19.832 18.477 18.247 18 16.5 18c-1.746 0-3.332.477-4.5 1.253"/>
        </svg>
      )
    }
  ];

  return (
    <>
      <Header title="Guide" />

      <div className="guide-main-container">
        <div className="guide-flex-content">
          <div className="guide-wrapper">
            <div className="guide-card-stack">
              {cards.map((card) => (
                  <div
                      key={card.id}
                      className="base-card guide-card"
                      onClick={() => handleCardClick(card.id)}


                >
                  <div className="card-content">
                    <h3 className="card-title card-title--with-description">
                      {card.title}
                    </h3>
                    <p className="card-description">
                      {card.description}
                    </p>
                  </div>
                  <div className="card-arrow-container">
                    <div className="card-arrow-circle">
                      {card.icon}
                    </div>
                  </div>
                </div>
              ))}
            </div>
          </div>
        </div>
      </div>

      <BottomNav />
    </>
  );
}