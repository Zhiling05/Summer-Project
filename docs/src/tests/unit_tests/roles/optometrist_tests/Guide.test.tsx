import { render, screen, fireEvent } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import GuideHome from '../../../../pages/optometrist/guide/GuideHome';

jest.mock('../../../../components/Header', () => ({ title }: { title: string }) => (
  <div data-testid="header">{title}</div>
));

jest.mock('../../../../components/BottomNav', () => () => (
  <div data-testid="bottom-nav">BottomNav</div>
));

const mockNavigate = jest.fn();
jest.mock('react-router-dom', () => {
  const original = jest.requireActual('react-router-dom');
  return {
    ...original,
    useNavigate: () => mockNavigate,
  };
});

describe('GuideHome Component', () => {
  test('renders both cards and header', () => {
    render(
      <MemoryRouter>
        <GuideHome />
      </MemoryRouter>
    );

    expect(screen.getByTestId('header')).toHaveTextContent('Guide');
    expect(screen.getByText('Reference Image Gallery')).toBeInTheDocument();
    expect(screen.getByText('App Tutorial for Optometrists')).toBeInTheDocument();
    expect(screen.getByTestId('bottom-nav')).toBeInTheDocument();
  });

  test('navigates to gallery when gallery card clicked', () => {
    render(
      <MemoryRouter>
        <GuideHome />
      </MemoryRouter>
    );

    fireEvent.click(screen.getByText('Reference Image Gallery'));
    expect(mockNavigate).toHaveBeenCalledWith('gallery');
  });

  test('navigates to tutorial when tutorial card clicked', () => {
    render(
      <MemoryRouter>
        <GuideHome />
      </MemoryRouter>
    );

    fireEvent.click(screen.getByText('App Tutorial for Optometrists'));
    expect(mockNavigate).toHaveBeenCalledWith('tutorial');
  });
});