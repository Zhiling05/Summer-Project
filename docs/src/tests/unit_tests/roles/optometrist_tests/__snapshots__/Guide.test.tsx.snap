// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Guide matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Optometrist Guide Page
    </h2>
  </div>
</DocumentFragment>
`;
